### PR TITLE
Use create for serviceAccount instead of enabled

### PR DESCRIPTION
**Pull request by kimxogus**
Setting service account name with `serviceAccount.create` is useful with aws-cdk's service account using IRSA

```typescript
  // create service account in cdk kubernetes manifest
  const sa = new eks.ServiceAccount(scope, 'verdaccio-sa', {
    cluster: ...,
    name: 'verdaccio',
  });

...

  new eks.HelmChart(scope, 'verdaccio', {
    ...
    values: {
      serviceAccount: {
        enabled: false,  // do not create service account in helm chart
        name: sa.serviceAccountName,
      }
    }
  }
```

### CODE REVIEW

File: charts/verdaccio/Chart.yaml
Comment (juanpicado) [inline charts/verdaccio/Chart.yaml:4]: Not sure whether this is just a patch, according what I read about this feature. I am waiting more opinions.

Comment (kimxogus) [reply charts/verdaccio/Chart.yaml:4]: I was a little bit confused here too. I'm fine with minor version bump

Comment (kimxogus) [reply charts/verdaccio/Chart.yaml:4]: I thought this might have been added by mistake because condition is already in `_helpers.tpl`, so I made patch bump

**Review by jfrancisco0 — COMMENTED** [review]
Hum, is this really "duplicated" in `_helpers.tpl`? It doesn't set `Values.serviceAccount.enabled` to `true`, I think, it just sets the name. So you're removing that check.

**Comment by kimxogus** [comment]
Well, I thought `.Values.serviceAccount.enabled` only used to define service account name and there's already default value, so I assumed that that might not be an intentioned behavior(As `.values.serviceAccount.enabled` check is already in `deployment.yaml`, it seems there's no reason another check is needed in `_helpers.tpl`).
Anyway, as it changed that behavior, I also agree that this is minor version bump. I changed it

**Comment by jfrancisco0** [comment]
I may be misreading this, but I think the flow on `_helpers.tpl` is:
- if `serviceAccount` is enabled, set the serviceAccount name to `Values.serviceAccount.name` if it's set, or `verdaccio.fullname` otherwise.
- if `serviceAccount` is not enabled, set the serviceAccount name to "default".

The serviceAccount will only be created if `Values.serviceAccount.enabled` is [true](https://github.com/verdaccio/charts/blob/master/charts/verdaccio/templates/service-account.yaml#L1), though.

If you eliminate the check, as proposed in this PR, you'll be adding a `serviceAccountName` that may not exist to the deployment, right?

**Comment by todpunk** [comment]
@jfrancisco0 is correct to my reading of the `_helpers.tpl` and the `service-account.yaml` files, and is a relatively common pattern. The serviceaccount name in the Deployment doesn't need to be set to default explicitly though.

If future-proofing for not-`default` service account namespaces in the future is desire, the if condition should not be removed, it should be extended to include the service-account name being set.

**Comment by kav** [comment]
Should we consider switching "enabled" to "create" as well here? I think that might clear up the confusion. It seems a bit off to set the name and enabled to false if you want it set but externally managed. It's certainly not going to hurt to have the serviceAccount name set to default if nothing is changed since that is the kubernetes default anyway.

**Comment by kimxogus** [comment]
I totally agree with switching to `create`, and I considered that writing this PR, but it should be major version bump and I didn't want to make breaking change, so I retained `enabled`.
Sorry if you are puzzled with the name `enabled`. For its purpose, `create` is right I guess.
@jfrancisco0 Yes! You're right.

**Comment by kimxogus** [comment]
@jfrancisco0 Yes

**Comment by kimxogus** [comment]
Is this ok to merge?

**Comment by juanpicado** [comment]
waiting for @jfrancisco0 , for me is OK, but not my field to be honest.

**Review by juanpicado — APPROVED** [review]

**Comment by jfrancisco0** [comment]
I listed some limitations of this in my comments, and honestly I'm not too familiar with the workings of the service account either. @todpunk would you like to weigh in?

**Review by jhonmike — APPROVED** [review]

**Review by jhonmike — COMMENTED** [review]
Oops, I saw now that there are really different validations, I don't think it's duplicated, one validates the name and the other validates the insertion of the attribute

**Comment by kimxogus** [comment]
@jhonmike @jfrancisco0 Then, should this go with `create` and bump major version?

**Comment by jfrancisco0** [comment]
@kimxogus I think something should change in `_helpers.tpl` too, so name of the serviceAccount doesn't depend on its creation by the chart.

**Comment by todpunk** [comment]
I agree with @jfrancisco0 's comment. If we do a `create` helpers needs some adjustment.

**Comment by kimxogus** [comment]
I changed to use `create`.
@jfrancisco0 @todpunk Can you tell more about what adjustments are needed? I referred to https://github.com/prometheus-community/helm-charts/blob/main/charts/prometheus-adapter/templates/_helpers.tpl but it seems to be the same here.

**Comment by juanpicado** [comment]
> I changed to use `create`.
> @jfrancisco0 @todpunk Can you tell more about what adjustments are needed? I referred to https://github.com/prometheus-community/helm-charts/blob/main/charts/prometheus-adapter/templates/_helpers.tpl but it seems to be the same here.

I plan to merge first https://github.com/verdaccio/charts/pull/52 which a minor change, please rebase when that happens, after that we can merge this if all involved are happy.

**Comment by jfrancisco0** [comment]
Could you just change the comment in line 73 of `values.yaml` (can't add the comment directly)?
`# If not set and enabled is true, a name is generated using the Chart's fullname template`
Enabled is now "create".

**Comment by kimxogus** [comment]
@juanpicado Rebased!
@jfrancisco0 Thanks for pointing this out, I missed that 😄  Changed to `create`

**Review by juanpicado — APPROVED** [review]
LGTM

**Review by jfrancisco0 — APPROVED** [review]
LGTM

**Pull request merged**
